databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -482,18 +625,18 @@ impl proto::val_server::Val for broker::DataBroker {
                                 match request {
                                     Some(req) => {
                                         match req.action {
-                                            Some(ProvidedActuation(_provided_actuation)) => {
-                                                if let Err(err) = response_stream_sender.send(Err(tonic::Status::new(tonic::Code::Unimplemented, "Unimplemented"))).await {
-                                                    debug!("Failed to send error response: {}", err);
-                                                }
-                                                break;
-                                            },
-                                            Some(PublishValuesRequest(publish_values_request)) => {
+                                           Some(PublishValuesRequest(publish_values_request)) => {
                                                 let response = publish_values(&broker, &publish_values_request).await;
                                                 if let Err(err) = response_stream_sender.send(Ok(response)).await
                                                 {
                                                     debug!("Failed to send response: {}", err);
                                                 }
+                                            },
+                                             Some(ProvideActuation(provided_actuation)) => {
+                                                if let Err(err) = provide_actuation(&broker, &provided_actuation, response_stream_sender).await {
+                                                    debug!("Failed to provide actuation: {}", err)
+                                                }
+                                                break;
                                             },
```

Comment:
Right now Errors are not properly propagated to the Provider - he won't know if registration fails (e.g. invalid signal) except of maybe due to the log. Since we are in a thread here, we can't return the err to the function body.

Reply:
Method `provide_actuation` btw was changed to return an Result<ProvideActuationResponse, tonic::Status> and send it to the sender. However, this still doesn't fix above mentioned problem. 

Should we even establish a connection in case the signal does not exist? (just to say 'Yeah, you got a channel now, but you will never receive any responses except of this one, telling you that you provided the wrong signal') Maybe some kind of pre-verification / pre-validation is needed.